tests/runtime/model_management/test_model_manager.py

```diff
@@ -1054,3 +1055,99 @@ def test_lazy_load_handles_temporary_errors():
                 assert manager._lazy_sync_timer is None
                 model = manager.retrieve_model(model_name)
                 assert model
+
+
+def test_lazy_load_true_local_models_dir_valid():
+    """When lazy_load_local_models is True and local_models_dir exists.
+    Check that the local_models_dir is pointing to the correct location
+    """
+
+    with TemporaryDirectory() as cache_dir:
+
+        ModelManager._ModelManager__instance = None
+        with temp_config(
+            {
+                "runtime": {
+                    "local_models_dir": cache_dir,
+                    "lazy_load_local_models": True,
+                }
+            },
+            merge_strategy="merge",
+        ):
+            MODEL_MANAGER = ModelManager()
+            assert len(MODEL_MANAGER.loaded_models) == 0
+            assert MODEL_MANAGER._local_models_dir == cache_dir
+
+
+def test_lazy_load_true_local_models_dir_invalid():
+    """When lazy_load_local_models is True and local_models_dir does not exist.
+    Raise ValueError with an appropriate message
+    """
+
+    with TemporaryDirectory() as cache_dir:
+
+        with pytest.raises(
+            ValueError,
+            match=(
+                "runtime.local_models_dir must be a valid path if set with"
+                " runtime.lazy_load_local_models"
```

Comment:
nit: can we check `Provided path` too?

Reply:
Do you mean update the patch to have the path: As part of this test, it will be an empty string to nothing will really change in the test match string